components/upload/interface.ts

```diff
@@ -36,6 +36,7 @@ export interface UploadFile<T = any> extends ProgressAriaProps {
   type?: string;
   xhr?: T;
   preview?: string;
+  metadata?: any;
```

Comment:
What problem does this feature solve?

Reply:
This allows us to create and pass around functions file objects of type `UploadFile`  that have extra application-specific metadata associated with the file without having other objects, which forces us to map the files to those objects wherever we want to use a file and its associated data.

Reply:
Just to give you a simplified example for use, we create the file name using a function that outputs the name of the file based on the application metadata of our file. We want to pass as a parameter of this function the `UploadFile` file only. Without this, we would need to track and pass the metadata in parallel to this and all functions that use the file and wherever the metadata is needed. This becomes hard to maintain when it could be all inside the same object.

Reply:
This is mostly a typescript issue since we can attach the property metadata to the object and it works, but the compiler will complain because metadata is not a defined field of the `UploadFile` type.

Reply:
Could we get some help with this reasonably simple PR? It is an enhancement purely based on TypeScript and does not affect any logical functionality.